basic/012_basic_program/getMode.js

```diff
@@ -0,0 +1,40 @@
+// Menentukan modus / nilai yang paling sering muncul
+
+/*
+ * Algoritma:
+ * case: terdapat sebuah array [5, 4, 5, 4, 5]
+ * 1. Petakan setiap nilai yang ada menjadi sebuah objek dengan nilai sebagai key dan jumlah kemunculannya sebagai value
+ *      -- cth: {4: 2, 5: 3}
+ *      -- angka 4 muncul 2 kali, angka 5 muncul 3 kali
+ * 2. Lakukan pengulangan pada objek tadi berdasarkan key lalu cek
+ *      -- jika value dari key itu lebih dari variabel max maka isi / timpa variabel max dengan valuenya,
+ *          lali isi / timpa array results menjadi array yang hanya memiliki 1 isi yang berisi key tadi
+ *      -- jika value dari key itu sama dengan variabel max maka tambah array results (push) dengan keynya
+ */
+
+function getMode(arr) {
+  const data = {};
+  arr.forEach((num) => {
+    if (data[num] === undefined) {
```

Comment:
Codacy ga bisa nerima `undefined`

Fixnya bisa gini
```suggestion
    if (!data[num]) {
```